### PR TITLE
MINOR: docs should point to latest version

### PR DESCRIPTION
**Pull request by mjsax**
*More detailed description of your change,
if necessary. The PR title and PR message become
the squashed commit message, so use a separate
comment to ping reviewers.*

*Summary of testing strategy (including rationale)
for the feature or bug fix. Unit and/or integration
tests are expected for any behaviour change and
system tests should be considered for larger changes.*

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by mjsax** [comment]
\cc @bbejeck @vvcephei

File: docs/documentation.html [hunk 1]
```diff
@@ -69,15 +69,15 @@ <h2><a id="security" href="#security">7. Security</a></h2>
     <h2><a id="connect" href="#connect">8. Kafka Connect</a></h2>
     <!--#include virtual="connect.html" -->
 
-    <h2><a id="streams" href="/11/documentation/streams">9. Kafka Streams</a></h2>
+    <h2><a id="streams" href="/streams">9. Kafka Streams</a></h2>
```
Comment (guozhangwang) [inline docs/documentation.html]: Will this be leading to `https://kafka.apache.org/streams/` which is a 404?

Comment (mjsax) [reply docs/documentation.html]: Think you are right -- I miss-understood the layout/nesting... Will update this.

**Comment by mjsax** [comment]
Updated this.

**Review by guozhangwang — APPROVED** [review]

**Pull request merged**